File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/sql/unique/UniqueTest.java
package com.raizlabs.android.dbflow.test.sql.unique;

import com.raizlabs.android.dbflow.sql.language.SQLite;
import com.raizlabs.android.dbflow.test.FlowTestCase;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

/**
 * Description:
 */
public class UniqueTest extends FlowTestCase {

    UniqueModel uniqueModel;

    @Before
    public void createModel() {
        uniqueModel = new UniqueModel();
        uniqueModel.uniqueName = "This is unique";
        uniqueModel.anotherUnique = "This should rollback";
        uniqueModel.sharedUnique = "This is unique to both combos";
    }

    @Test
    public void testUniqueReplacesExisting() {
        uniqueModel.anotherUnique = System.currentTimeMillis() + "";// guarantee unique for this test.
        uniqueModel.insert();
        assertTrue(uniqueModel.exists());

        uniqueModel.insert();
        assertEquals(1, SQLite.selectCountOf().from(UniqueModel.class).count());
    }
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/sql/unique/UniqueTest.kt
package com.raizlabs.android.dbflow.sql.unique

import com.raizlabs.android.dbflow.sql.language.SQLite
import com.raizlabs.android.dbflow.FlowTestCase
import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test

/**
 * Description:
 */
class UniqueTest : FlowTestCase() {

    lateinit var uniqueModel: UniqueModel

    @Before
    fun createModel() {
        uniqueModel = UniqueModel()
        uniqueModel.uniqueName = "This is unique"
        uniqueModel.anotherUnique = "This should rollback"
        uniqueModel.sharedUnique = "This is unique to both combos"
    }

    @Test
    fun testUniqueReplacesExisting() {
        uniqueModel.anotherUnique = System.currentTimeMillis().toString() + ""// guarantee unique for this test.
        uniqueModel.insert()
        assertTrue(uniqueModel.exists())

        uniqueModel.insert()
        assertEquals(1, SQLite.selectCountOf().from(UniqueModel::class.java).count())
    }
}
